tuning/data/data_processors.py

```diff
@@ -282,46 +282,43 @@ def __execute_filter_data_handler(self, raw_datasets, handler, **kwargs):
             kwargs["desc"] = handler.desc
         return raw_datasets.filter(handler.op, **kwargs)
 
-    def __execute_map_data_handler(
-        self, raw_datasets, handler, splitName, datasetName, **kwargs
-    ):
-        column_names = None
-        if hasattr(raw_datasets[splitName], "column_names"):
-            column_names = raw_datasets[splitName].column_names
-
-        # remove __content__ from all processing
-        if column_names and "__content__" in column_names:
-            column_names.remove("__content__")
+    def __execute_map_data_handler(self, raw_datasets, handler, datasetName, **kwargs):
+        """Apply handler.op to all splits in raw_datasets instead of just a single split."""
+        processed_ds = (
+            IterableDatasetDict()
+            if isinstance(raw_datasets, IterableDatasetDict)
+            else DatasetDict()
+        )
 
+        # set up kwargs for map
+        if not isinstance(raw_datasets, (IterableDatasetDict, IterableDataset)):
+            kwargs["desc"] = handler.desc
         if "remove_columns" not in kwargs:
             kwargs["remove_columns"] = None
-        if kwargs["remove_columns"] == "all":
-            if column_names is None:
-                logger.warning(
-                    "Could not infer column names from the dataset %s \
-                    unable to set `remove_columns` to all.\n \
-                    Please explicitly specify the column list or \
-                    use remove/select data handlers.",
-                    datasetName,
-                )
-            else:
-                kwargs["remove_columns"] = column_names
-
         if "fn_kwargs" not in kwargs:
             kwargs["fn_kwargs"] = {}
-
         kwargs["fn_kwargs"]["tokenizer"] = self.tokenizer
-        kwargs["fn_kwargs"]["column_names"] = column_names
 
-        # IterableDatasets doesn't support any description
-        if not isinstance(raw_datasets, (IterableDatasetDict or IterableDataset)):
-            kwargs["desc"] = handler.desc
+        for split_name, ds in raw_datasets.items():
+
+            if kwargs["remove_columns"] == "all":
+                column_names = getattr(ds, "column_names", None)
+                if column_names is None:
+                    logger.warning(
```

Comment:
This is not related to the PR changes but on a second thought can you make this an error and raise it 
this might end up doing damage to the user expercation if remove_columns is not adhered to.